modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/artifact/MavenClassLoaderModelLoader.java

```diff
@@ -70,15 +71,15 @@ public ClassLoaderModel load(File artifactFile, Map<String, Object> attributes,
     long stamp = lock.readLock();
     try {
       MavenConfiguration updatedMavenConfiguration = getMavenConfig();
-      if (!mavenRuntimeConfig.equals(updatedMavenConfiguration)) {
+      if (mavenRuntimeConfig == null || !Objects.equal(mavenRuntimeConfig, updatedMavenConfiguration)) {
```

Comment:
I think you can remove the null verificaton mavenRuntimeConfig == null. updatedMavenConfiguration will not be null, so it will enter whenever mavenRuntimeConfig is null (or there is another problem and mavenRuntimeConfig will be assigned null.